description/src/type_example/rust_value.rs

```diff
@@ -179,13 +190,13 @@ fn ty_example(
         scale_info::TypeDef::Sequence(def) => {
             // return a Vec with 2 elements:
             let inner_ty = transformer.resolve_type(def.type_param.id)?;
-            let item_code = ty_example(def.type_param.id, inner_ty, transformer)?; // todo!("Might need CompactMode::Expl")
+            let item_code = ty_example(def.type_param.id, inner_ty, transformer)?;
```

Comment:
Have you tested some compact encoded types yet? the TODOs are gone but nothing has changed, so I'm wondering how/if they are addressed :)

Reply:
Yes, the different handling of fields that are compact types and fields that just have `#[codec(compact)]`, is addresses here by investigating the `type_name`:
https://github.com/paritytech/scale-typegen/blob/recursive-derives/description/src/type_example/rust_value.rs#L244

I also added a test to confirm this works as expected: 
https://github.com/paritytech/scale-typegen/blob/recursive-derives/description/src/lib.rs#L227